src/chainstate/burn/db/sortdb.rs

```diff
@@ -1697,20 +1749,21 @@ impl <'a> SortitionDBConn <'a> {
                 stable_snapshot.block_height - MAX_NEIGHBOR_BLOCK_DELAY
             };
 
-        let mut last_consensus_hashes = HashMap::new();
+        let mut last_burn_block_hashes = HashMap::new();
         for height in oldest_height..chain_tip.block_height {
-            let ch = db_handle.get_consensus_at(height)?
-                .unwrap_or(ConsensusHash::empty());
-            last_consensus_hashes.insert(height, ch);
+            let ancestor_hash = SortitionDB::get_ancestor_snapshot(&self, height as u64, &chain_tip.sortition_id)?
```

Comment:
Would this loop be less expensive if the range was reversed?

Reply:
Yes, it would be much less expensive.  I'll go ahead and fix that.

Reply:
Do you think this is something that might be generally useful to support via the MARF interface (i.e., get parent Trie)?

Reply:
Possibly; we already store the data needed to do this.  But to make this class of operations go faster -- i.e. walking through a sequence of rows where each row has a parent/child relationship with another row -- the trick to speeding it up would be to simply put the parent's primary key in the child (e.g. `parent_sortition_id`) in order to obviate the need to touch the MARF at all.

Reply:
I think that it wouldn't make too much of a difference in speed -- doing a parent-child traversal from the MARF should just be an indexed lookup (without touching the blob data, which is more expensive to fetch from), so it'd likely just be one extra indexed SQLite query. Either way seems fine to me, but if down the line we want to do this in other MARF stores, then we should implement in the MARF.